pycbc/events/significance.py

```diff
@@ -0,0 +1,253 @@
+                parser.error(err_fmat.format(value, key))
+
+            if type_to_convert(value) not in allowed_values:
+                err_fmat = "Value {} of key {} is not in allowed values: {}"
+                parser.error(err_fmat.format(value, key, allowed_values))
+
+    # Are the functions/thresholds appropriate for the methods given?
+    methods = {}
+    # A method has been specified
+    for key_value in args.far_calculation_method:
+        key, value = tuple(key_value.split(':'))
+        methods[key] = value
+
+    # A function or threshold has been specified
+    function_or_thresh_given = []
+    for key_value in args.fit_function + args.fit_threshold:
+        key, _ = tuple(key_value.split(':'))
+        if key not in methods:
+            methods[key] = 'n_louder'
+        function_or_thresh_given.append(key)
+
+    for key, value in methods.items():
+        if value != 'trigger_fit' and key in function_or_thresh_given:
+            # Function/Threshold given for key not using trigger_fit method
+            parser.error("--fit-function and/or --fit-threshold given for "
+                         + key + " which has method " + value)
+        elif value == 'trigger_fit' and key not in function_or_thresh_given:
+            # Threshold not given for trigger_fit key
+            parser.error("Threshold required for key " + key)
+
+
+def digest_significance_options(combo_keys, args):
+    """
+    Read in information from the significance option group and ensure
+    it makes sense before putting into a dictionary
+
+    Parameters
+    ----------
+
+    combo_keys: list of strings
+        list of combinations of detectors which could be used as keys
+        for the options
+
+    args: parsed arguments
+        from argparse ArgumentParser parse_args()
+
+    Returns
+    -------
+    significance_dict: dictionary
+        Dictionary containing method, threshold and function for trigger fits
+        as appropriate
+    """
+
+    lists_to_unpack = [('method', args.far_calculation_method, str),
+                       ('function', args.fit_function, str),
+                       ('threshold', args.fit_threshold, float)]
+
+    # Set up the defaults
+    default_dict = {'method': 'n_louder', 'threshold': None, 'function': None}
+
+    # Unpack the string arguments into a standard-format dictionary
+    significance_dict = {}
+
+    # Set everything as a default to start with:
+    for key in combo_keys:
+        significance_dict[key] = copy.deepcopy(default_dict)
+
+    # Unpack everything from the arguments into the dictionary
+    for unpack_key, arg_to_unpack, conv_func in lists_to_unpack:
+        for key_value in arg_to_unpack:
+            key, value = tuple(key_value.split(':'))
+            if key not in significance_dict:
+                # This is a newly added key, not actually used by the code
+                # This is a warning not an exit, so we can reuse the same
```

Comment:
I can't quite understand the comment about reusing settings for multiple jobs. Is the idea that we have some global bunch of options for all possible ifo combinations and then each job may not have 'combo_keys' covering all of those combinations?

Reply:
That's right. It makes it a bit easier when setting up the config files, as we can reuse the value for the add_statmap arguments, where all combinations are used, but the individual coinc_statmap jobs dont use every combination

Reply:
OK .. so IMO the discussion of workflows / config files is a bit too indirectly related to what's happening here in the code to belong in a comment.
Suggest something like this.. 
`# Allow options for detector combos that are not actually used/required for a given job. Such options have no effect, but emit a warning for (eg) diagnostic checks`